modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java

```diff
@@ -423,6 +424,135 @@ public void testSendNotification() throws Exception {
+		resourcePermissionLocalService.addResourcePermission(
+			guestUser.getCompanyId(), parentObjectDefinition.getResourceName(),
+			ResourceConstants.SCOPE_COMPANY,
+			String.valueOf(guestUser.getCompanyId()), guestRole.getRoleId(),
+			ObjectActionKeys.ADD_OBJECT_ENTRY);
+
+		String body = RandomTestUtil.randomString();
+		String fromName = RandomTestUtil.randomString();
+		String subject = RandomTestUtil.randomString();
+
+		NotificationTemplate notificationTemplate =
+			notificationTemplateLocalService.addNotificationTemplate(
+				NotificationTemplateUtil.createNotificationContext(
+					TestPropsValues.getUser(), 0,
+					LocalizationUtil.updateLocalization(
+						LocalizedMapUtil.getLocalizedMap(
+							HashMapBuilder.put(
+								LanguageUtil.getLanguageId(LocaleUtil.US),
+								RandomTestUtil.randomString()
+							).put(
+								LanguageUtil.getLanguageId(LocaleUtil.FRANCE),
+								body
+							).build()),
+						null, "Body",
+						LanguageUtil.getLanguageId(LocaleUtil.US)),
+					RandomTestUtil.randomString(),
+					NotificationTemplateConstants.EDITOR_TYPE_RICH_TEXT,
+					Arrays.asList(
+						createNotificationRecipientSetting(
+							NotificationRecipientSettingConstants.NAME_FROM,
+							"[%CURRENT_USER_EMAIL_ADDRESS%]"),
+						createNotificationRecipientSetting(
+							NotificationRecipientSettingConstants.
+								NAME_FROM_NAME,
+							HashMapBuilder.put(
+								LocaleUtil.FRANCE, fromName
+							).put(
+								LocaleUtil.US, RandomTestUtil.randomString()
+							).build()),
+						createNotificationRecipientSetting(
+							NotificationRecipientSettingConstants.NAME_TO,
+							HashMapBuilder.put(
+								LocaleUtil.FRANCE, user1.getEmailAddress()
+							).put(
+								LocaleUtil.US, user2.getEmailAddress()
+							).build()),
+						createNotificationRecipientSetting(
+							NotificationRecipientSettingConstants.NAME_TO_TYPE,
+							NotificationRecipientConstants.TYPE_EMAIL),
+						createNotificationRecipientSetting(
+							NotificationRecipientSettingConstants.
+								NAME_USE_MOST_RELEVANT_LOCALE_FOR_GUEST_USERS,
+							StringPool.TRUE)),
+					LocalizationUtil.updateLocalization(
+						HashMapBuilder.put(
+							LocaleUtil.FRANCE, subject
+						).put(
+							LocaleUtil.US, RandomTestUtil.randomString()
+						).build(),
+						null, "Subject", "en_US"),
+					NotificationConstants.TYPE_EMAIL, Collections.emptyList()));
+
+		PermissionChecker originalPermissionChecker =
+			PermissionThreadLocal.getPermissionChecker();
+		String originalName = PrincipalThreadLocal.getName();
+
+		try {
+			PermissionThreadLocal.setPermissionChecker(
+				PermissionCheckerFactoryUtil.create(guestUser));
+			PrincipalThreadLocal.setName(guestUser.getUserId());
+
+			executeNotificationObjectAction(
+				new DefaultDTOConverterContext(
+					false, Collections.emptyMap(),
+					BaseNotificationTypeTest.dtoConverterRegistry, null,
+					LocaleUtil.fromLanguageId("fr_FR"), null, guestUser),
```

Comment:
@pedro-oliveira446 use `LocaleUtil.FRANCE` to follow the same pattern you used above.

Reply:
Hey @MarinhoFeliphe ,
I tried to use this, but it didn't work.